### ISSUE TITLE
Sporadic unit test failure testAppendFails

### ISSUE THREAD

**Issue by RaulGracia** [pravega/pravega#3292]
**Problem description**
Found test `AppendProcessorTest.testAppendFails` failing in a Jenkins build:
```
Error Message
org.mockito.exceptions.verification.NeverWantedButInvoked: 
dynamicLogger.incCounterValue(
    "segmentstore.segment.write_bytes_global",
    8L
);
Never wanted here:
-> at io.pravega.segmentstore.server.host.handler.AppendProcessorTest.testAppendFails(AppendProcessorTest.java:368)
But invoked here:
-> at io.pravega.segmentstore.server.host.handler.AppendProcessor.handleAppendResult(AppendProcessor.java:302)
Stacktrace
org.mockito.exceptions.verification.NeverWantedButInvoked: 
dynamicLogger.incCounterValue(
    "segmentstore.segment.write_bytes_global",
    8L
);
Never wanted here:
-> at io.pravega.segmentstore.server.host.handler.AppendProcessorTest.testAppendFails(AppendProcessorTest.java:368)
But invoked here:
-> at io.pravega.segmentstore.server.host.handler.AppendProcessor.handleAppendResult(AppendProcessor.java:302)

	at io.pravega.segmentstore.server.host.handler.AppendProcessorTest.testAppendFails(AppendProcessorTest.java:368)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
```

**Problem location**
`AppendProcessorTest.testAppendFails`

**Suggestions for an improvement**
Make the test to pass consistently.

**Comment by tkaitchuck** [comment]
This actually looks serious. I don't see how there is any non-determinism in this test. It doesn't even involve threads.

**Comment by tkaitchuck** [comment]
This seems to be the culprit: https://github.com/pravega/pravega/pull/3239
But what I don't get is how that assertion can fail, but the one a few lines up `verifyNoMoreInteractions` on the connection can not trigger as right above `AppendProcessor.java:302` is sends an ack on the connection.

**Comment by tkaitchuck** [comment]
It looks like to have have to find a way to either make DYNAMIC_LOGGER not shared between tests, or to verify the code without overriding static final members.

**Comment by kevinhan88** [comment]
@tkaitchuck @RaulGracia - since MetricsProvider.getDynamicLogger is static, we can change AppendProcessor: DYNAMIC_LOGGER = MetricsProvider.getDynamicLogger() non-static, with test visibility. Hence for production runtime DYNAMIC_LOGGER is static(shared) but for test code it can be overwritten per AppendProcessor instance. 
I'm sending out PR shortly.

**Comment by RaulGracia** [comment]
Fixed via #3292.
